dbt/adapters/base/impl.py

```diff
@@ -291,6 +293,27 @@ def set_macro_context_generator(
     ) -> None:
         self._macro_context_generator = macro_context_generator
 
+    @property
+    def behavior(self) -> Behavior:
+        return self._behavior
+
+    @behavior.setter
+    def behavior(self, flags: List[BehaviorFlag]) -> None:
+        flags.extend(self._behavior_extra)
+        try:
+            # we don't always get project flags, for example during `dbt debug`
+            self._behavior = Behavior(flags, self.config.flags)
+        except AttributeError:
+            # in that case, don't load any behavior to avoid unexpected defaults
+            self._behavior = Behavior([], {})
+
+    @property
+    def _behavior_extra(self) -> List[BehaviorFlag]:
```

Comment:
Not sure I have a great alternative here but instead of `_extra` could we be more specific? Maybe `_behavior_for_adapter` and then have a `_behavior_global`?

Reply:
I'll update this to `_behavior_flags`